### ISSUE TITLE
Missing files in new image after committing a container

### ISSUE THREAD

**Issue by oblitum** [containers/buildah#1868]
**Description**

I have an ArchLinux image which I used docker to build due to #1867. I've then saved a tar for it, loaded it into rootless podman and squashed it with buildah. The new squashed image is working, all tools I've installed in it are there. Now, I've run this new image rootless and upgraded the system inside it with `pacman -Syu --noconfirm --removemake && pacman -Sc --noconfirm`. It upgraded an AUR package, starship-bin. All is fine while the image is still live, and `/usr/bin/starship` still exists. Now when I exit the container, and commit it, when I run the committed image, `/usr/bin/starship` is not there anymore, which means the image got corrupted, because pacman list starship-bin as an installed package, but now it's missing a file.

**Output of `rpm -q buildah` or `apt list buildah`:**

```
❯ pacman -Qi buildah
Name            : buildah
Version         : 1.10.1-2
Description     : A tool which facilitates building OCI images
Architecture    : x86_64
URL             : https://github.com/containers/buildah
Licenses        : Apache
Groups          : None
Provides        : None
Depends On      : ostree  runc  skopeo  slirp4netns
Optional Deps   : None
Required By     : None
Optional For    : None
Conflicts With  : None
Replaces        : None
Installed Size  : 24.99 MiB
Packager        : Bartłomiej Piotrowski <bpiotrowski@archlinux.org>
Build Date      : Sun 01 Sep 2019 04:45:27 PM -03
Install Date    : Sun 01 Sep 2019 07:02:25 PM -03
Install Reason  : Explicitly installed
Install Script  : No
Validated By    : Signature
```

**Output of `buildah version`:**

```
❯ buildah version
Version:         1.10.1
Go Version:      go1.12.9
Image Spec:      1.0.1
Runtime Spec:    1.0.1-dev
CNI Spec:        0.4.0
libcni Version:  v0.7.1
Git Commit:      8c1c2c58
Built:           Sun Sep  1 16:45:49 2019
OS/Arch:         linux/amd64
```

**Output of `podman version` if reporting a `podman build` issue:**

```
❯ podman version
Version:            1.5.1
RemoteAPI Version:  1
Go Version:         go1.12.8
OS/Arch:            linux/amd64
```

**Output of `cat /etc/*release`:**

```
❯ cat /etc/*release
Arch Linux release
LSB_VERSION=1.4
DISTRIB_ID=Arch
DISTRIB_RELEASE=rolling
DISTRIB_DESCRIPTION="Arch Linux"
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="0;36"
HOME_URL="https://www.archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://bugs.archlinux.org/"
LOGO=archlinux
```

**Output of `uname -a`:**

```
❯ uname -a
Linux leibniz 5.2.14-arch2-1-ARCH #1 SMP PREEMPT Thu Sep 12 10:42:38 UTC 2019 x86_64 GNU/Linux
```

**Output of `cat /etc/containers/storage.conf`:**

```
❯ cat ~/.config/containers/storage.conf
[storage]
  driver = "overlay"
  runroot = "/run/user/1000"
  graphroot = "/home/francisco/.local/share/containers/storage"
  [storage.options]
    size = ""
    remap-uids = ""
    remap-gids = ""
    ignore_chown_errors = ""
    remap-user = ""
    remap-group = ""
    ostree_repo = ""
    skip_mount_home = ""
    mount_program = "/usr/bin/fuse-overlayfs"
    mountopt = ""
    [storage.options.thinpool]
      autoextend_percent = ""
      autoextend_threshold = ""
      basesize = ""
      blocksize = ""
      directlvm_device = ""
      directlvm_device_force = ""
      fs = ""
      log_level = ""
      min_free_space = ""
      mkfsarg = ""
      mountopt = ""
      use_deferred_deletion = ""
      use_deferred_removal = ""
      xfs_nospace_max_retries = ""
```

**Comment by oblitum** [comment]
This is the command output when I commit the just run archlinux container to a new archlinux image:

```
❯ podman commit archlinux archlinux
Getting image source signatures
Copying blob fb2e53f6804a skipped: already exists
Copying blob 4ec1da521752 done
Copying config d77e96cd6c done
Writing manifest to image destination
Storing signatures
d77e96cd6c418e11fc8274cd583acb758a56769953adbd2c2142185ee42b9736
```

**Comment by TomSweeneyRedHat** [comment]
@oblitum Thanks for the problem report.  Just as a quick test if you've the time, can you try rerunning your test, but this time when you do the commit don't use the same name?  i.e. `podman commit archlinux myarchlinux`?

It's the first thing that comes to top of my mind, thanks.

**Comment by oblitum** [comment]
I tried, problem isn't due to container/image name, same behavior. I just did the same steps with docker and found no issues. I just loaded the tar into docker, run, upgrade, exit, commit, run, all there, no missing files. I think I'll have to mix docker, buildah and podman in my workflow when I go about to do upgrades. It's quite convoluted. I have to load my image in docker, do an upgrade to it, then save it to a tar, then load it into rootless podman, and squash it with buildah.

**Comment by rhatdan** [comment]
Are you running root or rootless. Could you see if everything works when running podman/buildah as root.

**Comment by oblitum** [comment]
I'm running podman rootless. Running podman as root on issue #1867 fixed that problem, but it may have been due because my root user isn't configured to use overlayfs. It also caused me to run out of disk space when I tried to run it as root for this current issue. I simply went to docker to fix my problem instead of setting overlayfs to root.

I'm sorry but I'm not interested in trying this as root again, as it's being quite burdensome, image is quite large, etc. I think I'll simply settle to operating upgrades and such operations in docker, and them simply saving to a tar and loading the image in rootless podman whenever I need to run it rootless.
